app/src/main/java/at/tacticaldevc/oat/utils/SMSCom.java

```diff
@@ -1,7 +1,88 @@
+        ensureStringIsValid(featureName, "name of disabled feature");
+
+        SmsManager smsManager = fetchSMSManager();
+        smsManager.sendTextMessage(phoneNumber, null, String.format(context.getString(R.string.oat_sms_message_error_feature_disabled), featureName), null, null);
+    }
+
+    /**
+     * Send a reply to let the User know, that the requested feature does not exist
+     *
+     * @param context     the {@link Context} of the Application
+     * @param phoneNumber the phone number that requested the non-existent feature
+     * @param featureName the name of the missing feature
+     */
+    public static void replyErrorSMS_FeatureNotFound(Context context, String phoneNumber, String featureName) {
+        ensureNotNull(context, "Application Context");
+        ensurePhoneNumberIsValid(phoneNumber, "phone number");
+        ensureStringIsValid(featureName, "feature name");
+
+        SmsManager smsManager = fetchSMSManager();
+        smsManager.sendTextMessage(phoneNumber, null, String.format(context.getString(R.string.oat_sms_message_error_feature_not_found), featureName), null, null);
+    }
+
+    public static void replyErrorSMS_InvalidPassword(Context context, String phoneNumber, String invalidPassword) {
+        ensureNotNull(context, "Application Context");
+        ensurePhoneNumberIsValid(phoneNumber, "phone number");
+
+        SmsManager smsManager = fetchSMSManager();
+        smsManager.sendTextMessage(phoneNumber, null, String.format(context.getString(R.string.oat_sms_message_error_invalid_password), invalidPassword), null, null);
+    }
+
+    /**
+     * Reply with the given GPS Position of the Device.
+     *
+     * @param context     the {@link Context} of the Application
+     * @param phoneNumber the phone number that asked for the GPS Position of the Device
+     * @param location    the current location of the Device
+     */
+    public static void replyFetchGPSPosition(Context context, String phoneNumber, Location location) {
+        ensureNotNull(context, "Application Context");
+        ensurePhoneNumberIsValid(phoneNumber, "target phone number");
+        ensureNotNull(location, "current location");
+
+        SmsManager smsManager = fetchSMSManager();
+        smsManager.sendTextMessage(phoneNumber, null, String.format(context.getString(R.string.oat_sms_message_send_gps_position), location.getLatitude(), location.getLongitude()), null, null);
+    }
+
+    /**
+     * @return the SMSManager associated with the default Subscription or null, if no valid subscription could be found
+     */
+    private static SmsManager fetchSMSManager() {
+        int subscriptionID = SmsManager.getDefaultSmsSubscriptionId();
+        if (!(subscriptionID == SubscriptionManager.INVALID_SUBSCRIPTION_ID)) {
+            return SmsManager.getSmsManagerForSubscriptionId(subscriptionID);
+        }
+        // Error handling missing
+        return null;
```

Comment:
OATException.forLibraryError()

Reply:
I created a new Exception:
OATApplicationException.forNoSMSSubscriptionFound()
This Error occurs if a phone has no Subscription to send SMS, as such it is not a Library Error.